images/metering-ansible-operator/roles/meteringconfig/library/k8s_event.py

```diff
@@ -0,0 +1,329 @@
+        argumentSpec.update(EVENT_ARG_SPEC)
+        return argumentSpec
+
+    def execute_module(self):
+        """ Module execution """
+        #auth
+        self.client = self.get_api_client()
+
+        # retrieve params
+        message = self.params.get("message")
+        reason = self.params.get("reason")
+        reporting_component = self.params.get("reportingComponent")
+        event_type = self.params.get("type")
+        source = self.params.get("source")
+
+        definition = defaultdict(defaultdict)
+
+        meta = definition["metadata"]
+        meta["name"] = self.params.get("name")
+        meta["namespace"] = self.params.get("namespace")
+
+        involved_obj = self.params.get("involvedObject")
+        print(involved_obj["kind"], involved_obj["namespace"], involved_obj["name"], involved_obj["apiVersion"])
+
+        resource = self.find_resource(
+            "Event", "v1", fail=True)
+
+        prior_event = None
+        try:
+            prior_event = resource.get(
+                name=meta["name"],
+                namespace=meta["namespace"])
+
+        except openshift.dynamic.exceptions.NotFoundError:
+            pass
+
+        # event counter
+        prior_count = 1
+        print("the count is no event exists %i" % prior_count)
+        now = datetime.datetime.now()
+        rfc = kubernetes.config.dateutil.format_rfc3339(now)
+        first_timestamp = rfc
+        last_timestamp = rfc
+
+        if prior_event and prior_event["reason"] == reason:
+            prior_count = prior_event["count"] + 1
+            print("If event exists and reason is the same %i" % prior_count)
+            first_timestamp = prior_event["firstTimestamp"]
+            print("the value of the firstTimestamp event exists same reason is", first_timestamp)
+            last_timestamp = rfc
+            print("the value of the lastTimestamp event exists same reason is", last_timestamp)
+
+        # handle involvedObject
+        involved_object_resource = self.find_resource(
+            involved_obj["kind"], "v1", fail=True)
+
+        if involved_object_resource:
+            try:
+                api_involved_object = resource.get(
+                    name=involved_obj["name"], namespace=involved_obj["namespace"])
+
+                involved_obj["uid"] = api_involved_object["metadata"]["uid"]
+                involved_obj["resourceVersion"] = api_involved_object["metadata"]["resourceVersion"]
+
+            except openshift.dynamic.exceptions.NotFoundError:
+                pass
+
+        event = {
+            "apiVersion": "v1",
+            "count": prior_count,
+            "eventTime": None,
+            "firstTimestamp": first_timestamp,
+            "involvedObject": involved_obj,
+            "kind": "Event",
+            "lastTimestamp": last_timestamp,
+            "message": message,
+            "metadata": {"name": meta["name"], "namespace": meta["namespace"]},
```

Comment:
I think this would be easier to read if the `metadata` key was multi-lined:
```python
            "metadata": {
                "name": meta["name"],
                "namespace": meta["namespace"],
             },
```